src/internal/m365/collection/drive/permission.go

```diff
@@ -385,7 +393,9 @@ func UpdateLinkShares(
 
 		newLS, err := upils.PostItemLinkShareUpdate(ictx, driveID, itemID, lsbody)
 		if graph.IsErrUsersCannotBeResolved(err) {
+			oldLinkShareIDToNewID.Store(ls.ID, "") // empty to signify that we could not restore
```

Comment:
Nit: a const might be easier to grok than a comment.
```suggestion
const nonRestorablePermission = ""
// ...
			oldLinkShareIDToNewID.Store(ls.ID, nonRestorablePermission)
```